ui/lib/core/addon/components/replication-page.js

```diff
@@ -20,128 +23,123 @@ import { task } from 'ember-concurrency';
     @model={{cluster}}
     />
  * ```
- * @param {Object} cluster=null - An Ember data object that is pulled from the Ember Cluster Model.
+ * @param {Object} model=null - An Ember data object that is pulled from the Ember Cluster Model.
  */
 
 const MODE = {
   dr: 'Disaster Recovery',
   performance: 'Performance',
 };
 
-export default Component.extend({
-  layout,
-  store: service(),
-  router: service(),
-  reindexingDetails: null,
-  didReceiveAttrs() {
-    this._super(arguments);
-    this.getReplicationModeStatus.perform();
-  },
-  getReplicationModeStatus: task(function* () {
-    let resp;
-    const { replicationMode } = this.model;
+export default class ReplicationPage extends Component {
+  @service store;
+  @service router;
+  @tracked reindexingDetails = null;
 
+  @action onModeUpdate(evt, replicationMode) {
+    // Called on did-insert and did-update
+    this.getReplicationModeStatus.perform(replicationMode);
+  }
+
+  @task
```

Comment:
does the `waiter` defined above behave differently/work better than using `waitFor` from `@ember/test-waiters`?

Reply:
I think they should be mostly the same, but the docs for `@ember/test-waiters` is literally nonexistent 😞 I'm happy to update this to `waitFor` if that's easier to read

Reply:
No need to update! Was just curious 😊

Reply:
too late! 😝